src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -40,12 +40,24 @@ StatelessComponent.prototype.render = function() {
 
 function warnIfInvalidElement(Component, element) {
   if (__DEV__) {
-    warning(
-      element === null || element === false || ReactElement.isValidElement(element),
-      '%s(...): A valid React element (or null) must be returned. You may have ' +
-      'returned undefined, an array or some other invalid object.',
-      Component.displayName || Component.name || 'Component'
-    );
+    var componentName = Component.displayName || Component.name || 'Component';
+    if (element === undefined && Component._isMockFunction) {
+      warning(
+        false,
+        '%s(...): Stateless functional components must return a valid React element, ' +
+        'but %s appears to be a mocked function which returned `undefined`. Consider ' +
+        'disabling mocking (jest.autoMockOff or jest.unmock) for this component or ' +
```

Comment:
Should we leave "jest" out of this? There are other libraries that people use that do mocking. For example, even you didn't use jest mocking in your test.